esm, doc: fix and improve specifier resolution warning (#42252)
nodejs/node: doc/api/esm.md

> Stability: 1 - Experimental

> > **Note: Do not rely on this flag. We plan to remove it once the

Author: I think I made the same suggestion (to use passive voice) months ago to @Trott and he pointed me to the style guide: https://docs.microsoft.com/en-us/style-guide/grammar/person. Basically, “we” is discouraged, but it’s still better than the passive voice. Whoever is the authority on style should tell me how this should be phrased.

> > **Note: Do not rely on this flag. We plan to remove it once the

Reviewer: In general, the Microsoft Style Guide prefers active voice, but suggests passive voice for some situations. In particular, it suggests that warnings and notifications might be better as passive voice. I would consider this a warning/notification, so passive voice is likely appropriate.

Ref: https://docs.microsoft.com/en-us/style-guide/grammar/verbs#active-and-passive-voice

> > **Note: Do not rely on this flag. We plan to remove it once the

Author: I feel like I grew up being taught that technical documentation should be in the passive voice, but clearly the modern style is active. If anyone has an idea for a replacement for “we” I would consider that, but I think “we” is better than the passive voice.

> > **Note: Do not rely on this flag. We plan to remove it once the

Reviewer: > I feel like I grew up being taught that technical documentation should be in the passive voice, but clearly the modern style is active.

Yeah, pretty much. The passive voice makes things sound formal, so published documentation adopted that for a long time. At some point, people decided that documentation (especially on the web) should be friendly and relatable rather than formal and authoritative-sounding, and active voice was adopted. Additionally, passive voice is supposedly harder to understand for a lot of folks whose first language is not English. But neither passive voice nor active voice are "wrong".

> If anyone has an idea for a replacement for “we” I would consider that, but I think “we” is better than the passive voice.

```markdown
> **Note: Do not rely on this flag. It will get the boot once the
```

> > **Note: Do not rely on this flag. We plan to remove it once the
> > [Loaders API][] has advanced to the point that equivalent functionality can
> > be achieved via custom loaders.**

Reviewer: ```suggestion
> This flag will be removed when the [Loaders API][] offers equivalent
> functionality via custom loaders.
```

Reviewer: That's fine. You can ignore my suggestion. Or maybe incorporate just a part of it and get rid of `Note:` and maybe the bold text?

```suggestion
> Do not rely on this flag. We plan to remove it once the
> [Loaders API][] has advanced to the point that equivalent functionality can
> be achieved via custom loaders.
```

Again, though, feel free to ignore it. My preference is less typographical variance and more concise wording, but every situation is also unique and has other considerations.

Author: The Loaders warning is in bold and starts with `Note:`. I want these to match, either both with `Note:`/bold or both without `Note:`/bold. Which is your preference?

Reviewer: My preference is both without, but that is a non-blocking preference, so smashing the "Resolve conversation" button on this and not changing a thing is totally an option.

Author: I think it was @bmeck who put the loaders warning in bold, so giving him fair warning here that I’m removing `Note:` and bold from the Loaders warning unless he objects.
